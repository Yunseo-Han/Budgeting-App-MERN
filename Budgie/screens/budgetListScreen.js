import React, {useEffect, useState} from "react";

import {
  SafeAreaView,
  ScrollView,
  StatusBar,
  StyleSheet,
  Modal, 
  Text,
  View,
  TouchableOpacity,
  TouchableHighlight,
  TextInput,
  KeyboardAvoidingView,
  Dimensions
} from 'react-native';

import { buttonGrey, addButtonBlue } from '../budgieColors';
import DatePicker from 'react-native-date-picker';
import Swipeable from 'react-native-swipeable-row';

// REALM
import {useMemo} from 'react';
import BudgetContext, { Budget } from "../models/Budget";
import { ObjectId } from "bson";



const deviceHeight = Dimensions.get('window').height;
const deviceWidth = Dimensions.get('window').width;

var isStartingDate = false;
var isEndingDate = false;


export const BudgetListScreen = ({ navigation }) => {
  const { useRealm, useQuery, RealmProvider } = BudgetContext;
  const realm = useRealm();
  const result = useQuery("Budget");
  const budgets = useMemo(() => result.sorted("startDate"), [result]);

  // Use states
  const [budgetItems, setBudgetItems] = useState([]);
  // const [budgetContainer, startBudgetContainer] = useState(BudgetContainer)
  const currentIdString = "";


  // Budget Modal
  const [budgetModalVisible, setBudgetModalVisible] = useState(false);

  
  function constructor(props) {
    this.onPress = this.onPress.bind(this);
  }

  // REALM
  function handleAddBudget(startDate, endDate, targetSpending) {
    targetSpending = parseFloat(targetSpending);
    let newBudget;
    realm.write(() => {
      newBudget = realm.create("Budget", new Budget({startDate, endDate, targetSpending}));
    });

    let i = 1;
    budgets.forEach(element => {
      console.log("Budget " + i);
      console.log(element._id.toString());
      console.log(element.startDate);
      console.log(element.endDate);
      console.log(element.targetSpending);
      console.log("\n");
      i++;
    });
  }

  function handleDeleteBudget(idString) {
    let id = ObjectId(idString);
    let budToDel = realm.objects("Budget").filtered("_id == $0", id)[0];
    realm.write(() => {
      realm.delete(budToDel);
    });
  }

  function handleDeleteAll() {
    realm.write(() => {
      // Delete all objects from the realm.
      realm.deleteAll();
    });
  }


  
  // Components
  const BudgetPreviewButton = ({startDate, endDate, spending, saving, idString}) => {

    // Handle functions   ******* used to be outside the component
    function pressedBudgetPreviewButton(idString) {
      navigation.navigate('Budget Details', {
        idString : idString
      });
    }

    const deleteButton = <TouchableHighlight style={styles.deleteButton} onPress ={() => handleDeleteBudget(idString)}><Text style={{paddingLeft: 20}}>Delete</Text></TouchableHighlight>

    return (
      <Swipeable rightButtons={[deleteButton]}>
        <TouchableOpacity onPress={() => pressedBudgetPreviewButton(idString)} style={styles.roundedButton}>
          <Text style={styles.importantText}> {startDate.toLocaleDateString() + " - " + endDate.toLocaleDateString()} </Text>
          <View style = {{alignContent: 'flex-end', maxWidth: '60%'}}>
            <Text> {'Limit: $' + spending.toFixed(2)} </Text>
            <Text> {'Remaining: $' + saving.toFixed(2)} </Text>
          </View>
        </TouchableOpacity>
      </Swipeable>
    );
  }




  const AddNewBudgetModal = () => {
    
    const [startingDate, setStartingDate] = React.useState(new Date());
    const [endingDate, setEndingDate] = React.useState(new Date())
    const [budgetLimit, setBudgetLimit] = React.useState(0);
    const [idString, setIdString] = React.useState("DEFAULT_ID_STRING");

    // Date picker data
    const [date, setDate] = useState(new Date())
    // Open date picker
    const [open, setOpen] = useState(false)
    // "Are we setting starting or ending?"" flag


    function addNewBudget() {
      console.log(startingDate, endingDate, budgetLimit);
      handleAddBudget(startingDate, endingDate, budgetLimit);
      setStartingDate("");
      setEndingDate("");
      setBudgetLimit("");
      setBudgetModalVisible(false);
    }

    function addStartingDate() {
      setOpen(true)
      isStartingDate = true
    }

    function addEndingDate() {
      setOpen(true)
      isEndingDate = true
    }

    return (
      <Modal
        animationType="slide"
        transparent={true}
        visible={budgetModalVisible}> 
        <View style={styles.centeredView}>
        <View style={styles.modalView}>
          <KeyboardAvoidingView behavior={Platform.OS === "ios" ? "padding" : "height"}>
            <View style={{flexDirection: 'row', alignSelf: 'stretch', justifyContent:'space-between', marginTop: 5}}>
              <Text style={styles.titleText}>Add Budget</Text>
              <TouchableOpacity style={styles.cancelButton} onPress={()=>setBudgetModalVisible(false)}>
                <Text>CANCEL</Text>
              </TouchableOpacity>
            </View>

            <View>
              <Text style={styles.textInputTitle}>Budget Limit</Text>
              <TextInput
                style={styles.textInputBox}
                keyboardType={'decimal-pad'}
                onChangeText={newLimit => setBudgetLimit(newLimit)}
              />
            </View>

            <View>
              <Text style={styles.textInputTitle} >Starting Date</Text>
              <TouchableOpacity style={styles.textInputBox} onPress={addStartingDate}>
                <Text>{startingDate.toLocaleDateString()}</Text>
              </TouchableOpacity>
            </View>
            
            <View>
              <Text style={styles.textInputTitle} >Ending Date</Text>
              <TouchableOpacity style={styles.textInputBox} onPress={addEndingDate}>
                <Text>{endingDate.toLocaleDateString()}</Text>
              </TouchableOpacity>
            </View>
            
            
            <TouchableOpacity style={styles.addBudgetButton} onPress={addNewBudget} >
              <Text>Add Budget</Text>
            </TouchableOpacity>

          </KeyboardAvoidingView>
          <DatePicker
            modal
            open={open}
            date={date}
            mode={"date"}
            textColor={addButtonBlue}
            onConfirm={(date) => {
              if (isStartingDate) {
                setStartingDate(date)
                isStartingDate = false
              } else if (isEndingDate) {
                setEndingDate(date)
                isEndingDate = false
              } 
              setOpen(false)
              
            }}
            onCancel={() => {
              setOpen(false)
              isStartingDate = false
              isEndingDate = false
            }}
          />
        </View>
        </View>
      </Modal>
    );
  }

    
  // Screen
  return (
    <SafeAreaView>
      <View style={{width: deviceWidth, height: deviceHeight}}>
      <StatusBar barStyle={'dark-content'}/>
        <ScrollView contentInsetAdjustmentBehavior="automatic">

          <View style={{flexDirection:'row', justifyContent:'space-between', paddingVertical: 10}}>
            <Text style={styles.titleText}> Budgets </Text>
            <TouchableOpacity style={styles.addButton} onPress={()=>setBudgetModalVisible(true)}>
              <Text>ADD</Text>
            </TouchableOpacity>
          </View>

          <AddNewBudgetModal/>

          <View>
            {
              budgets.map((item) => (
                <BudgetPreviewButton
                  key={item._id}
                  startDate={item.startDate}
                  endDate={item.endDate}
                  spending={item.targetSpending}
                  saving={item.targetSpending-item.totalSpending}
                  idString={item._id.toString()}
                />
              ))
            }
          </View>

        <TouchableOpacity style={styles.addButton} onPress={()=> handleDeleteAll()}>
            <Text>NUKE</Text>
        </TouchableOpacity>

          
        </ScrollView>
      </View>
    </SafeAreaView>
  );
};



const styles = StyleSheet.create({
  roundedButton: {
    backgroundColor: buttonGrey,
    borderRadius: 10,
    paddingVertical: 15,
    paddingHorizontal: 20,
    justifyContent: 'space-between',
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 10,
    marginHorizontal: 20,
  },

  titleText: {
    fontSize: 30,
    fontWeight: 'bold',
    marginVertical: 10,
    marginHorizontal: 10,
  },
  
  importantText: {
    fontWeight: 'bold',
    maxWidth: '70%',
  },

  budgetInput: {
    height: 40,
    margin: 12,
    borderWidth: 1,
    padding: 10,
  },

  addButton: {
    backgroundColor: addButtonBlue,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 20,
    width: 60,
    height: 40,
    marginTop: 10,
    marginRight: 10,
  },

  setBudgetContainer: {
    justifyContent: 'center',
    backgroundColor: 'white',
    width: 350,
    height: 400,
    alignItems: 'center',
    borderRadius: 30,
    borderColor: 'grey',
    borderWidth: 1,
    marginHorizontal: 10,
    marginVertical: 10,
    zIndex: 3,
    elevation: 3,
  },

  textInputTitle: {
    marginLeft: 10,
    color: 'grey',
    marginTop: 10,
    alignContent: 'flex-start',
  },

  textInputBox: {
    borderRadius: 20,
    borderColor: 'grey',
    borderWidth: 1,
    justifyContent: 'center',
    marginTop: 5, 
    paddingHorizontal: 10,
    height: 40,
    width: 300,
  },

  addBudgetButton: {
    backgroundColor: addButtonBlue,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 30,
    paddingVertical: 10,
    paddingHorizontal: 30,
    marginVertical: 20,
    marginHorizontal: 30,
  },


  cancelButton: {
    backgroundColor: buttonGrey,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 20,
    width: 70,
    height: 40,
    marginTop: 10,
    marginRight: 10,
  },

  modalView: {
    margin: 10,
    backgroundColor: "white",
    borderRadius: 10,
    paddingHorizontal: 35,
    paddingVertical: 10,
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5
  },

  centeredView: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    marginTop: 22,
  },

  deleteButton: {
    backgroundColor: '#FF6961',
    borderBottomLeftRadius: 10,
    borderTopLeftRadius: 10, 
    width: 200,
    marginRight: 10,
    height: 60,
    // paddingVertical: 15,
    // paddingHorizontal: 20,
    justifyContent: 'center',
    alignItems: 'flex-start',
    // marginBottom: 10,
    // marginHorizontal: 20,
  }
});