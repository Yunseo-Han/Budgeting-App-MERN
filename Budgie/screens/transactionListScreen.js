import React, {Component, useState} from 'react';
import {
    SafeAreaView,
    ScrollView,
    StatusBar,
    StyleSheet,
    Image,
    Text,
    useColorScheme,
    View,
    Modal,
    Pressable,
    TouchableOpacity,
    Button,
    KeyboardAvoidingView,
    TouchableHighlight,
    TextInput,
  } from 'react-native';

import { buttonGrey, addButtonBlue } from '../budgieColors';
import { Dimensions } from "react-native";
import Swipeable from 'react-native-swipeable-row';

  //REALM
import {useMemo} from 'react';
import BudgetContext, { Budget, Category, Transaction} from "../models/Budget";
import { ObjectId } from "bson";

const screenWidth = Dimensions.get("window").width;




export const TransactionListScreen = ({ navigation, route }) => {

    // const [spendingContainer, setSpendingContainer] = useState(SpendingContainer)
    const [spendingItems, setSpendingItems] = useState([]);
    

    const {catIdString, budIdString} = route.params;

    // add spending modal 
    const [spendingModalVisible, setSpendingModalVisible] = useState(false);
   
    const screenWidth = Dimensions.get("window").width;


    //REALM
    const { useRealm, useQuery, RealmProvider } = BudgetContext;
    const realm = useRealm();

    const budId = ObjectId(budIdString);
    const currentBudget = realm.objects("Budget").filtered("_id == $0", budId)[0];

    const catID = ObjectId(catIdString);
    const currentCat = realm.objects("Category").filtered("_id == $0", catID)[0];

    // Reversing the transactions list so that it puts the newest transactions at the top.
    const result = useQuery("Category").filtered("_id == $0", catID)[0].transactions;
    const reversedTxs = useMemo(() => result.sorted("date", true), [result]);

    function constructor(props) {
      this.onPress = this.onPress.bind(this);
    }

    function handleAddTransaction(spendingName, date, amount) {
      let newTrans;
      let amt = parseFloat(parseFloat(amount).toFixed(2));
      if(isNaN(amt)) {
        console.log("Amount entered is not a number.");
        return;
      }
      if(amt < 0) {
        console.log("Amount entered must be greater than 0.");
        return;
      }
      if(spendingName == "") {
        spendingName = 'Transaction #' + (currentCat.txCounter+1);
      }
      realm.write(() => {
        newTrans = realm.create("Transaction", new Transaction({name: spendingName, date: date, amount: amt}));
        currentCat.transactions.push(newTrans);
        currentCat.transactionSum += amt;
        currentCat.txCounter += 1;
        currentBudget.totalSpending += currentCat.transactionSum;
      });
    }

    function handleDeleteTransaction(txIdString) {
      let id = ObjectId(txIdString);
      let txToDel = realm.objects("Transaction").filtered("_id == $0", id)[0];
      realm.write(() => {
        let diff = txToDel.amount;
        currentCat.transactionSum = currentCat.transactionSum - diff;
        currentBudget.totalSpending = currentBudget.totalSpending - diff;
        realm.delete(txToDel);
      });
    }

    function transactionsExist() {
      if(currentCat.transactions.length > 0) {
        return true;
      }
      return false;
    }

    const Spending = ({ title, amount, date, txIdString }) => {

      const deleteButton = <TouchableHighlight style={styles.deleteButton2} onPress ={() => handleDeleteTransaction(txIdString)}><Text style={{paddingLeft: 20}}>Delete</Text></TouchableHighlight>

      return( 
          <Swipeable rightButtons={[deleteButton]}>
            <View style = {styles.roundedButton}>
              <View style = {{alignContent: 'flex-start', maxWidth: '70%'}}>
                <Text style = {{fontWeight: 'bold'}}> {title} </Text>
                <Text> {date.toLocaleDateString()} </Text>
              </View>

              <View style = {{alignContent: 'flex-end', maxWidth: '30%'}}>
                <Text style = {styles.spendingAmount}> ${amount} </Text>
              </View>
            </View>
          </Swipeable>
        );
    };

    const ModalAddSpending = () => {
      const [spendingName, setSpendingName] = React.useState("");
      const [spending, setSpending] = React.useState(0);
      const [date, setDate] = React.useState(new Date());
      // const [amount, setAmount] = React.useState(0);
      
      
      function addTransaction(){
        handleAddTransaction(spendingName, date, spending);
        setSpendingName("");
        setDate("");
        setSpending(0);
        setSpendingModalVisible(false);
      }

      return (
        <Modal
        animationType="slide"
        transparent={true}
        visible={spendingModalVisible}
        >
          <View style={styles.centeredView}>
            <View style={styles.modalView}>
                <View style={{flexDirection: 'row', alignSelf: 'stretch', justifyContent:'space-between', marginTop: 5}}>
                  <Text style={styles.sectionTitle}>New Spending</Text>
                  <TouchableOpacity style={styles.cancelButton} onPress={() =>setSpendingModalVisible(false)}>
                    <Text>CANCEL</Text>
                  </TouchableOpacity>
                </View>
      
                <View>
                  <Text style={styles.textInputTitle}>Transaction Name</Text>
                  <TextInput
                    style={styles.textInputBox}
                    onChangeText={spendingName => setSpendingName(spendingName)}
                  />
                </View>
      
                <View>
                  <Text style={styles.textInputTitle}>Amount</Text>
                  <TextInput
                    style={styles.textInputBox}
                    keyboardType={'decimal-pad'}
                    onChangeText={newSpending => setSpending(newSpending)}
                  />
                </View>
      
                <TouchableOpacity style={styles.addBudgetButton} onPress={addTransaction}>
                  <Text>Add Spending</Text>
                </TouchableOpacity>
      
            </View>
          </View>
        </Modal>

      );
    }


    return(
      <SafeAreaView style={[styles.container, {backgroundColor: 'white'}]}>
      <StatusBar barStyle='dark-content'/>
      <ScrollView contentInsetAdjustmentBehavior="automatic">

      <View style = {{flexDirection:'row', justifyContent:'space-between', paddingVertical: 10}}>
        <Text style = {styles.titleText}> {currentCat.name}</Text>  
        <TouchableOpacity style={styles.deleteButton} onPress={() => handleDeleteCategory()}>  
          <Text>Delete Category</Text>
        </TouchableOpacity>
      </View>
        
      <View>
          {
              reversedTxs.map((item) => (
                <Spending
                  key = {item._id}
                  title = {item.name}
                  date = {item.date}   
                  amount = {item.amount.toFixed(2)}
                  txIdString = {item._id.toString()}
                />
              ))
          }
      </View>

      
      <View style={{height: 70}}/>
      
      <ModalAddSpending/>
    </ScrollView>

    <TouchableOpacity style={styles.addButton} onPress={()=>setSpendingModalVisible(true)}>
      <Text style={styles.plusSymbol}>+</Text>
    </TouchableOpacity>
    
  </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    height: "100%",
    paddingHorizontal : 10,
    paddingVertical : 20,
  },

  roundedButton: {
    backgroundColor: buttonGrey,
    borderRadius: 10,
    paddingVertical: 15,
    paddingHorizontal: 20,
    justifyContent: 'space-between',
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 10,
    marginHorizontal: 20,
  },

  spendingAmount: {
    fontWeight: 'bold',
    color:'red',
  },

  addCategoryButton: {
    backgroundColor: addButtonBlue,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 30,
    paddingVertical: 10,
    paddingHorizontal: 30,
    marginVertical: 20,
    marginHorizontal: 50,
  },

  buttonText: {
    fontSize: 15,
    fontWeight: 'bold',
    maxWidth: '70%',
  },

  centeredView: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    marginTop: 22,
  },

  modalView: {
    margin: 10,
    backgroundColor: "white",
    borderRadius: 10,
    paddingHorizontal: 35,
    paddingVertical: 10,
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5
  },

  sectionTitle:{
    fontSize: 20,
    fontWeight: 'bold',
    marginVertical: 10,
  },

  cancelButton: {
    backgroundColor: buttonGrey,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 20,
    width: 70,
    height: 40,
    marginTop: 10,
    marginRight: 10,
},

textInputTitle: {
  marginLeft: 10,
  color: 'grey',
  marginTop: 10,
  alignContent: 'flex-start',
},

textInputBox: {
  borderRadius: 20,
  borderColor: 'grey',
  borderWidth: 1,
  justifyContent: 'center',
  marginTop: 5, 
  paddingHorizontal: 10,
  height: 40,
  width: 300,
},

addBudgetButton: {
  backgroundColor: addButtonBlue,
  justifyContent: 'center',
  alignItems: 'center',
  borderRadius: 30,
  paddingVertical: 10,
  paddingHorizontal: 30,
  marginVertical: 20,
  marginHorizontal: 30,
},

sectionText: {
  fontSize: 20,
  fontWeight: 'bold',
  marginVertical : 10,
  marginHorizontal : 10,
},

addButton: {
  backgroundColor: addButtonBlue,
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'flex-end',
    bottom: 30,
    right: 30,
    borderRadius: 30,
    width: 60,
    height: 60,
    marginTop: 10,
    marginRight: 10,
    position: 'absolute',
    opacity: 0.9,
    shadowColor: 'rgba(0,0,0, .4)', // IOS
    shadowOffset: { height: 1, width: 1 }, // IOS
    shadowOpacity: 0.5, // IOS
    shadowRadius: 4, //IOS
    elevation: 2, // Android
},

centerAddSymbol: {
  justifyContent: 'center',
  alignItems: 'center',
  flex: 1
},

plusSymbol:{
  fontSize: 30,
  color: 'white',
  // fontWeight: 'bold',
  // paddingVertical: 10,
},

titleText: {
  fontSize: 30,
  fontWeight: 'bold',
  marginVertical: 10,
  marginHorizontal: 10,
},

deleteButton: {
  backgroundColor: 'tomato',
  justifyContent: 'center',
  alignItems: 'center',
  borderRadius: 30,
  paddingVertical: 10,
  paddingHorizontal: 10,
  marginVertical: 10,
  marginHorizontal: 10,
  shadowColor: 'rgba(0,0,0, .4)', // IOS
  shadowOffset: { height: 1, width: 1 }, // IOS
  shadowOpacity: 0.5, // IOS
  shadowRadius: 4, //IOS
  elevation: 2 // Android
},

deleteButton2: {
  backgroundColor: 'tomato',
  borderBottomLeftRadius: 10,
  borderTopLeftRadius: 10, 
  width: 200,
  marginRight: 10,
  height: 60,
  // paddingVertical: 15,
  // paddingHorizontal: 20,
  justifyContent: 'center',
  alignItems: 'flex-start',
  // marginBottom: 10,
  // marginHorizontal: 20,
}, 
});