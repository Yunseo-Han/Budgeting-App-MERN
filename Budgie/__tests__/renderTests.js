import React from 'react';
import renderer from 'react-test-renderer';
import {BudgetContainer, BudgetPreviewButton, BudgetListScreen} from '../screens/budgetListScreen';
import {BudgetScreen, Category, ModalAddCategory} from "../screens/budgetScreen" 

test('renders correctly', () => {
    const tree = renderer.create(<BudgetContainer/>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
    const tree = renderer.create(<BudgetPreviewButton/>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
    const tree = renderer.create(<BudgetListScreen/>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
    const tree = renderer.create(<BudgetScreen/>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
    const tree = renderer.create(<Category/>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
    const tree = renderer.create(<ModalAddCategory/>).toJSON();
    expect(tree).toMatchSnapshot();
});