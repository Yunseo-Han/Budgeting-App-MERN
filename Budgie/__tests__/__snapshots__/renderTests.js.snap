// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "height": "100%",
        "paddingHorizontal": 10,
        "paddingVertical": 20,
      },
    ]
  }
>
  <RCTScrollView
    contentInsetAdjustmentBehavior="automatic"
  >
    <View>
      <View
        style={
          Object {
            "alignContent": "center",
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://reactnative.dev/docs/assets/p_cat2.png",
            }
          }
          style={
            Object {
              "alignSelf": "center",
              "height": 200,
              "width": 200,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "borderTopColor": "#E3E3E3",
            "borderTopWidth": 2,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
              "marginHorizontal": 20,
              "marginVertical": 10,
            }
          }
        >
           Budget Categories 
        </Text>
        <View
          alignContent="center"
          flexDirection="row"
          justifyContent="flex-start"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#E3E3E3",
                "borderRadius": 10,
                "flex": 3,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginBottom": 10,
                "marginHorizontal": 10,
                "paddingHorizontal": 20,
                "paddingVertical": 15,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                  "maxWidth": "50%",
                }
              }
            >
               
              Rent
               
            </Text>
            <View
              style={
                Object {
                  "maxWidth": "50%",
                }
              }
            >
              <Text>
                 
                $1500
                 
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "white",
                "borderColor": "#E3E3E3",
                "borderRadius": 20,
                "borderWidth": 5,
                "height": 60,
                "marginBottom": 10,
                "marginRight": 10,
                "opacity": 1,
                "width": 60,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "marginHorizontal": 10,
                    "marginVertical": 15,
                  }
                }
              >
                ADD
              </Text>
            </View>
          </View>
        </View>
        <View
          alignContent="center"
          flexDirection="row"
          justifyContent="flex-start"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#E3E3E3",
                "borderRadius": 10,
                "flex": 3,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginBottom": 10,
                "marginHorizontal": 10,
                "paddingHorizontal": 20,
                "paddingVertical": 15,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                  "maxWidth": "50%",
                }
              }
            >
               
              Rent
               
            </Text>
            <View
              style={
                Object {
                  "maxWidth": "50%",
                }
              }
            >
              <Text>
                 
                $1500
                 
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "white",
                "borderColor": "#E3E3E3",
                "borderRadius": 20,
                "borderWidth": 5,
                "height": 60,
                "marginBottom": 10,
                "marginRight": 10,
                "opacity": 1,
                "width": 60,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "marginHorizontal": 10,
                    "marginVertical": 15,
                  }
                }
              >
                ADD
              </Text>
            </View>
          </View>
        </View>
        <View
          alignContent="center"
          flexDirection="row"
          justifyContent="flex-start"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#E3E3E3",
                "borderRadius": 10,
                "flex": 3,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginBottom": 10,
                "marginHorizontal": 10,
                "paddingHorizontal": 20,
                "paddingVertical": 15,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                  "maxWidth": "50%",
                }
              }
            >
               
              Rent
               
            </Text>
            <View
              style={
                Object {
                  "maxWidth": "50%",
                }
              }
            >
              <Text>
                 
                $1500
                 
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "white",
                "borderColor": "#E3E3E3",
                "borderRadius": 20,
                "borderWidth": 5,
                "height": 60,
                "marginBottom": 10,
                "marginRight": 10,
                "opacity": 1,
                "width": 60,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "marginHorizontal": 10,
                    "marginVertical": 15,
                  }
                }
              >
                ADD
              </Text>
            </View>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "stretch",
              "alignSelf": "center",
              "backgroundColor": "#E3E3E3",
              "borderRadius": 10,
              "marginBottom": 30,
              "marginTop": 20,
              "opacity": 1,
              "paddingHorizontal": 90,
            }
          }
        >
          <Text
            onPress={[Function]}
            style={
              Object {
                "fontSize": 20,
                "fontWeight": "bold",
                "paddingVertical": 10,
              }
            }
          >
            New Category
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderTopColor": "#E3E3E3",
            "borderTopWidth": 2,
            "marginHorizontal": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
              "marginVertical": 10,
            }
          }
        >
           New Spendings 
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#E3E3E3",
              "borderRadius": 10,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 10,
              "paddingHorizontal": 20,
              "paddingVertical": 15,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                  "maxWidth": "50%",
                }
              }
            >
               
              Nordstrom
               
            </Text>
            <Text>
               
              4/6/2022
               
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "red",
                  "fontWeight": "bold",
                }
              }
            >
               
              $500.00
               
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#E3E3E3",
              "borderRadius": 10,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 10,
              "paddingHorizontal": 20,
              "paddingVertical": 15,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                  "maxWidth": "50%",
                }
              }
            >
               
              Best Buy
               
            </Text>
            <Text>
               
              4/6/2022
               
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "red",
                  "fontWeight": "bold",
                }
              }
            >
               
              $50.00
               
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#E3E3E3",
              "borderRadius": 10,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 10,
              "paddingHorizontal": 20,
              "paddingVertical": 15,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                  "maxWidth": "50%",
                }
              }
            >
               
              Target
               
            </Text>
            <Text>
               
              4/6/2022
               
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "red",
                  "fontWeight": "bold",
                }
              }
            >
               
              $54.00
               
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "marginTop": 22,
          }
        }
      >
        <Modal
          animationType="slide"
          hardwareAccelerated={false}
          onRequestClose={[Function]}
          transparent={true}
          visible={false}
        />
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
